src/test/java/com/williamfiset/algorithms/strings/BoyerMooreStringSearchTest.java

```diff
@@ -35,12 +42,78 @@ public void shouldReturnOneOccurrence() {
   }
 
   @Test
-  public void shouldReturnMultiplyOccurrences() {
+  public void shouldReturnMultipleOccurrences() {
+    assertThat(underTest.findOccurrences("SAAT TE", "TE")).containsExactly(5);
     assertThat(
             underTest.findOccurrences("Sample text for testing the Boyer-Moore algorithm.", "te"))
         .containsExactly(7, 16);
     assertThat(underTest.findOccurrences("Sample text for testing the Boyer-Moore algorithm.", " "))
         .containsExactly(6, 11, 15, 23, 27, 39);
+    assertThat(underTest.findOccurrences("AABAACAADAABAABA", "AABA")).containsExactly(0, 9, 12);
+    assertThat(underTest.findOccurrences("AAAAAAA", "AA")).containsExactly(0, 1, 2, 3, 4, 5);
+  }
+
+  @Test
+  public void shouldReturnEmptyForPatternLengthLargerThenText() {
+    assertThat(underTest.findOccurrences("This is a Test Text", "This is a test Pattern"))
+        .isEmpty();
+  }
+
+  @Test
+  public void shouldReturnDynamicString() {
+    int runLength = random.nextInt(MAX_ITERATION) + 1;
+    for (int run = 0; run < runLength; run++) {
+      int upperCharText = random.nextInt(26);
```

Comment:
Suggestion: To increase the number of matches restrict the alphabet size. For instance, make the text and pattern only from the alphabet {'A' and 'B'} rather than {'A', 'B', ..., 'Z'}

Reply:
Added code to limit the number of characters.